lib/posix-process/process.h

```diff
@@ -87,7 +87,20 @@ struct posix_thread {
 	/* TODO: Mutex */
 };
 
+extern struct posix_process *pid_process[TIDMAP_SIZE];
+
+#define uk_pprocess_foreach(_p)						\
+	for (int _j = 1, _i = 0; _i != ARRAY_SIZE(pid_process);		\
+		_j = !_j, _i++)						\
+			for ((_p) = pid_process[_i]; _j; _j = !_j)	\
+				if ((_p))
+
+#define uk_pprocess_foreach_pthread(_proc, _pthread, _pthreadn)		\
```

Comment:
Nitpick. Just occurred to me, but all `foreach`s we have end either in `foreach` or `reverse`, while the components through which they iterate precede said `foreach`... except for `uk_sched_foreach_thread` only AFAICT. Should we perhaps also name this `uk_pprocess_pthread_foreach` instead? And rename `uk_sched_foreach_thread` to `uk_sched_thread_foreach` the first occasion we get in whatever future PR may touch this part of the code?

Reply:
Yeah okay, will update